tests/conftest.py

```diff
@@ -0,0 +1,75 @@
+import pytest
+import httpx
+from unittest import mock
+
+# from contextlib import asynccontextmanager
+import pytest_asyncio
+from fakeredis import FakeStrictRedis
+from fastapi.testclient import TestClient
+from server.app import _get_redis_conn
+
+
+def pytest_addoption(parser):
+    parser.addoption(
+        "--use-openai",
+        action="store_true",
+        default=False,
+        help="run tests that require OpenAI key",
+    )
+    parser.addoption(
+        "--use-server",
+        action="store_true",
+        default=False,
+        help="run tests that require running adala server",
+    )
+
+
+def pytest_configure(config):
+    config.addinivalue_line("markers", "use_openai: mark test as requiring OpenAI key")
+    config.addinivalue_line(
+        "markers", "use_server: mark test as requiring running adala server"
+    )
+
+
+def pytest_collection_modifyitems(config, items):
```

Comment:
Why not just use the regular `pytest -m` for those markers, rather than make our own flags?

Reply:
Mind giving an example of how that'd work? The real reason is I've never done this before, looked up how to do it and this way made sense :)

Reply:
sure yeah, you can target different marks using e.g. `pytest -m use_openai` or `pytest -m "use_openai or use_server".
This is more standard from what I've seen in the past, but definitely not the hill I'll die on